doc/explanation/external-builder.md

```diff
@@ -0,0 +1,5 @@
+# External builder
+
+In order to pre-bootstrap LXD, Microk8s snaps to Juju, external VMs are spanwed and then run with
+cloud-init script that installs required components. Then, it is snapshot to work around limitation
```

Comment:
Nitpick: Add "the" in two places: 
- "script that installs the required components" 
- "it is snapshot to work around the limitation"